src/Cli/dotnet/commands/dotnet-help/HelpCommand.cs

```diff
@@ -99,18 +92,18 @@ public int Execute()
                 Reporter.Error.WriteLine(
                     string.Format(
                         LocalizableStrings.CommandDoesNotExist,
-                        _parseResult.GetValue(HelpCommandParser.Argument)).Red());
+                        helpArgs).Red());
                 Reporter.Output.WriteLine(HelpUsageText.UsageText);
                 return 1;
             }
         }
 
-        private bool TryGetDocsLink(string commandName, out string docsLink)
+        private bool TryGetDocsLink(string[] command, out string docsLink)
         {
-            var command = Parser.GetBuiltInCommand(commandName);
-            if (command != null && command as DocumentedCommand != null)
+            var parsedCommand = Parser.Instance.Parse(["dotnet", .. command]);
```

Comment:
This is the magic sauce - we can ask S.CL to resolve the command that the args map to without any guessing at all.